Project/core/src/main/java/Model/AbstractCell.java

```diff
@@ -75,10 +75,28 @@ public void addToken(Token newToken) {
 
     /**
      * Removes a token from the cell.
-     * 
+     *
      * @param token The token to remove from the cell.
      */
     public void removeToken(Token token) {
         tokens.remove(token);
     }
+
+    public Footstep getFootstep() {
+        for (Token token : tokens) {
+            if (token.getClass().equals(Footstep.class)) {
+                return (Footstep) token;
+            }
+        }
+        throw new IllegalStateException("No Footstep found in the cell");
+    }
+
+    public boolean containsFootstep() {
+        for (Token token : tokens) {
+            if (token.getClass().equals(Footstep.class)) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
These changes will conflict with PR #37, and I think maybe Fox's version would be preferred, imo.

Reply:
We will merge this and handle it in Fox's PR